R/request.R

```diff
@@ -9,13 +9,20 @@ join_url <- function(url, endpoint) {
 #' @importFrom httr RETRY
 do_request <- function(url, params) {
   # don't retry in case of certain status codes
-  res <- httr::RETRY("GET", url,
-    query = params, http_headers,
+  res <- httr::RETRY("GET",
+    url,
+    query = params,
+    http_headers,
+    httr::authenticate("epidata", get_auth_key()),
```

Comment:
praise: good that the API key isn't a query parameter; I think the special treatment is supposed to be best practice to prevent logging keys all over the place.

question: Should we use `httr::handle`s?  `?httr::authenticate` recommends this, but I don't think its examples actually use it.  Instead, maybe we could mirror the tests [here](https://github.com/r-lib/httr/commit/f0f8de394cc55b66643a48c9f3a156213ea35f86)?  I'm not sure if it's that necessary for us if we re-pass the auth key every time (no chance of it trying to reuse a mistyped key again and again if the user has fixed the env var / option, hopefully).

Reply:
Well that's frustrating. Why make a function if you're going to recommend working around it?

I tried using handle, but it gave me an error in `parse_url`. I don't really want to spend more time on this, until we actually get a fix request.

```r
  handle <- httr::handle(url)
  res <- httr::RETRY("GET",
    handle = handle,
    query = params,
    terminate_on = c(400, 401, 403, 405, 414, 500),
    http_headers,
    httr::authenticate("epidata", get_auth_key())
  )
```